src/lib/phdi-building-blocks/phdi_building_blocks/fhir.py

```diff
@@ -327,29 +350,42 @@ def fhir_server_get(
     return response
 
 
-def log_fhir_server_error(status_code: int) -> None:
+def log_fhir_server_error(status_code: int, batch_entry_index: int = None) -> None:
     """Given an HTTP status code from a FHIR server's response, log the specified error.
 
     :param status_code: Status code returned by a FHIR server
     """
+
+    batch_decorator = ""
+    if batch_entry_index is not None:
+        batch_decorator = f"in entry number {batch_entry_index} of batch "
+
     if status_code == 401:
-        logging.error("FHIR SERVER ERROR - Status Code 401: Failed to authenticate.")
+        logging.error(
+            f"FHIR SERVER ERROR {batch_decorator}- Status Code 401: Failed to "
+            + "authenticate."
+        )
 
     elif status_code == 403:
         logging.error(
-            "FHIR SERVER ERROR - Status Code 403: User does not have permission to make that request."  # noqa
+            f"FHIR SERVER ERROR {batch_decorator}- Status Code 403: User does not "
+            + "have permission to make that request."
         )
 
     elif status_code == 404:
         logging.error(
-            "FHIR SERVER ERROR - Status Code 404: Server or requested data not found."
+            f"FHIR SERVER ERROR {batch_decorator}- Status Code 404: Server or "
+            + "requested data not found."
         )
 
     elif status_code == 410:
         logging.error(
-            "FHIR SERVER ERROR - Status Code 410: Server has deleted this cached data."
+            f"FHIR SERVER ERROR {batch_decorator}- Status Code 410: Server has "
+            + "deleted this cached data."
         )
 
     elif str(status_code).startswith(("4", "5")):
-        error_message = f"FHIR SERVER ERROR - Status code {status_code}"
+        error_message = (
+            f"FHIR SERVER ERROR {batch_decorator}- Status code {status_code}"
+        )
         logging.error(error_message)
```

Comment:
Nice job integrating `batch_decorator` and leaving the old behavior unchanged. Cool!